botocore/docs/utils.py

```diff
@@ -124,3 +125,49 @@ def document_auto_populated_param(self, event_name, section, **kwargs):
             section = section.get_section('structure-value')
             if self.name in section.available_sections:
                 section.delete_section(self.name)
+
+
+class HideParamFromOperations(object):
+    """Hides a single parameter from multiple operations.
+
+    This method will remove a parameter from documentation and from
+    examples. This method is typically used for things that are
+    automatically populated because a user would be unable to provide
+    a value (e.g., a checksum of a serialized XML request body)."""
+    def __init__(self, service, parameter_name, operation_names):
+        """
+        :type parameter_name: str
+        :param parameter_name: Name of the parameter to modify.
+
+        :type operation_names: list
+        :param operation_names: Operation names to modify.
+        """
+        self._parameter_name = parameter_name
+        self._event_list = {}
+        # Build up a hash of relevant event names.
```

Comment:
I used a hash here because we already know exactly what type of event is is without the need for an additional string check. It's also faster (O(1)) to look in a hash for a specific event name. Considering there are a ton of documentation events, I thought this was a good technical reason for the hash with a precomputed event type ("params" vs "example").

Did you have a reason to not use a hash?